experiments/book-surprisal/ocr_surprisal.py

```
@@ -0,0 +1,68 @@
+        for pnum in row["pages"].split(","):
+            n=f"image{int(pnum):05d}.txt"
+            if n not in pages: continue
+            pg=pages[n]["first_pass"]; sv=sorted(pg["surprisal"]); pct=sv[int(0.95*len(sv))]
+            h=word_hits(row["word"], pg["text"], pg["offsets"], pg["surprisal"], pct); art_hits+=h; arts+=1 if h else 0
+random.seed(7); rnd_hits=[]
+for n,pg in pages.items():
+    fp=pg["first_pass"]; sv=sorted(fp["surprisal"]); pct=sv[int(0.95*len(sv))]
+    words=[w for w in re.findall(r"[A-Za-z]{3,}", fp["text"])]
+    for w in random.sample(words, min(3,len(words))): rnd_hits+=word_hits(w, fp["text"], fp["offsets"], fp["surprisal"], pct)[:1]
+res["artifact_words_found"]=arts; res["artifact_hit_rate_at_page_p95"]=round(sum(art_hits)/max(1,len(art_hits)),3); res["artifact_occurrences"]=len(art_hits)
+res["random_word_hit_rate_at_page_p95"]=round(sum(rnd_hits)/max(1,len(rnd_hits)),3); res["random_occurrences"]=len(rnd_hits)
```

Ruff (0.16.3)

[error] Multiple statements on one line (semicolon)

(E702)